src/elements/common/annotator-context/withAnnotations.tsx

```diff
@@ -75,20 +77,39 @@ export default function withAnnotations<P extends object>(
             annotator.emit('annotations_active_set', id);
         };
 
-        emitRemoveEvent = (id: string) => {
+        emitAnnotationRemoveEvent = (id: string, isStartEvent = false) => {
             const { annotator } = this;
 
             if (!annotator) {
                 return;
             }
 
-            annotator.emit('annotations_remove', id);
+            // Event name does not include "sidebar" namespace because of backwards compatibility with Preview
+            const event = isStartEvent ? 'annotations_remove_start' : 'annotations_remove';
+
+            annotator.emit(event, id);
+        };
+
+        emitAnnotationUpdateEvent = (annotation: Object, isStartEvent = false) => {
+            const { annotator } = this;
+
+            if (!annotator) {
+                return;
+            }
+
+            const event = isStartEvent ? 'sidebar.annotations_update_start' : 'sidebar.annotations_update';
```

Comment:
Why do we need the `sidebar` prefix here but not elsewhere?

Reply:
This one is a bit tricky. I would love to use the sidebar namespace for all events that originate from sidebar, however it needs to be backwards compatible with Preview integration and I can't change the event name for removing the annotation (`annotations_remove`). I also can't add the common namespace for all events from outside of sidebar, because of the same reason - Preview emits events `annotations_active_change` and `annotations_create`. Current implementation also uses a different event name for changing the active annotation. When the change is done in sidebar it's `annotations_active_set` and when in Preview it's `annotations_active_change`. This is a good candidate for refactoring in future when there's more time so that the naming is unified for all parties.